torch/_dynamo/eval_frame.py

```diff
@@ -554,6 +554,62 @@ def toy_example(a, b):
     )
 
 
+@dataclasses.dataclass
+class ExplainOutput:
+    """
+    This is the output of :func:`torch._dynamo.explain()
+    There is no reason to create this class directly
+    """
+
+    graph_count: int
+    graph_break_count: int
+    op_count: int
+    out_guards: List[_guards.Guard]
+    graphs: List[torch.fx.GraphModule]
+    ops_per_graph: List[torch.fx.Node]
+    break_reasons: List[Any]  # TODO: Not sure if graph break is a type
+    compile_times: str
+
+    def __str__(self):
+        output = f"Graph Count: {self.graph_count}\n"
+        output += f"Graph Break Count: {self.graph_break_count}\n"
+        output += f"Op Count: {self.op_count}\n"
+
+        output += "Break Reasons:\n"
+        for idx, break_reason in enumerate(self.break_reasons):
+            output += f"  Break Reason {idx+1}:\n"
+            output += f"    Reason: {break_reason.reason}\n"
+            output += "    User Stack:\n"
+            for frame_summary in break_reason.user_stack:
+                output += f"      {frame_summary}\n"
+
+        output += "Ops per Graph:\n"
+        for idx, ops in enumerate(self.ops_per_graph):
+            output += f"  Ops {idx+1}:\n"
+            for op in ops:
+                output += f"    {op}\n"
+
+        output += "Out Guards:\n"
+        for i, guard in enumerate(self.out_guards):
+            output += f"  Guard {i+1}:\n"
+            output += f"    Name: {guard.name}\n"
+            output += f"    Source: {guard.source}\n"
+            output += f"    Create Function: {guard.create_fn}\n"
+            output += f"    Is Volatile: {guard.is_volatile}\n"
+            output += f"    Guard Types: {guard.guard_types}\n"
+            output += f"    Code List: {guard.code_list}\n"
+            output += f"    Object Weakref: {guard.obj_weakref}\n"
+            output += f"    Guarded Class Weakref: {guard.guarded_class_weakref}\n"
+
+        # This looks really ugly so far so not sure how I feel about adding it
+        # output += "Graphs:\n"
+        # for graph in self.graphs:
+        #     output += "  " + graph.print_readable() + "\n"
+
```

Comment:
Maybe a print_graphs() func?

Reply:
But what should it print? I feel like a graph is too big and frankly not necessary, am considering just getting rid of this entirely